ms-utils/math/linear_algebra/cuda/CudaDenseDecomp.hh

```diff
@@ -51,6 +51,9 @@ class CudaDenseDecomp : public CholeskyLdu, public CudaIfUtils
         virtual ~CudaDenseDecomp();
         /// @brief Decomposes the admittance matrix [A].
         virtual void Decompose(double *A, int n);
+        /// @brief Decomposes only the rows of the admittance matrix [A] given by the vector
+        ///        argument r.
+        virtual void Decompose(double *A, int n, std::vector<int>& r) { CholeskyLdu::Decompose(A, n, r); }
```

Comment:
We probably need unit test for this new function, also move the implementation to an 'inline' at the bottom of the file, for consistency with the repo.

Reply:
This function is only a pass-through that prevents CudaDenseDecomp::Decompose(double, int) from shadowing CholeskyLdu::Decompose(double, int, std::vector<int>). Is it really necessary to effectively test the same function twice?

Alternatively, do we _want_ CholeskyLdu::Decompose(double, int, std::vector<int>) to be unavailable here? I could make this function private and unimplemented to achieve that more explicitly.

Reply:
Yes we want to test it, even though it seems trivial.  We strive for 100% line coverage, except for lines that are too impractical or impossible to reach.  This one line of code may seem trivial to test, but it is still possible to mess it up, by getting arguments to the called function wrong, or calling the wrong function, etc..  This unit test is not just to test your update, but also to provide regression tests on future updates by someone else.  No we don't want to make the CholeskyLdu::Decompose private, because the solver Gunns class needs to call it in some configurations.

Reply:
Added a unit test. I copied it straight from CholeskyLdu.